buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiTask.java

```diff
@@ -0,0 +1,218 @@
+        return includeCore;
+    }
+
+    @Input
+    public ListProperty<String> getIncludeXpack() {
+        return includeXpack;
+    }
+
+    @SkipWhenEmpty
+    @InputFiles
+    public FileTree getInputDir() {
+        xpackPatternSet.setIncludes(includeXpack.get().stream().map(prefix -> prefix + "*/**").collect(Collectors.toList()));
+        ConfigurableFileCollection fileCollection = getProject().files(xpackConfig.getAsFileTree().matching(xpackPatternSet));
+        if (BuildParams.isInternal()) {
+            corePatternSet.setIncludes(includeCore.get().stream().map(prefix -> prefix + "*/**").collect(Collectors.toList()));
+            fileCollection.plus(coreConfig.getAsFileTree().matching(corePatternSet));
+        } else {
+            fileCollection.plus(coreConfig);
+        }
+        return "spec".equals(getTestOrSpec()) && projectHasYamlRestTests()
+            || includeCore.get().isEmpty() == false
+            || includeXpack.get().isEmpty() == false ? fileCollection.getAsFileTree() : null;
+    }
+
+    @OutputDirectory
+    public File getOutputDir() {
+        return new File(getTestSourceSet().getOutput().getResourcesDir(), copyTo);
+    }
+
+    @TaskAction
+    void copy() {
+        Project project = getProject();
+        if (BuildParams.isInternal()) {
+            logger.info("Rest {} for project [{}] will be copied to the test resources.", getTestOrSpec(), project.getPath());
+            project.copy(c -> {
+                c.from(coreConfig.getSingleFile());
+                c.into(getOutputDir());
+                c.include(corePatternSet.getIncludes());
+            });
+
+        } else {
+            logger.info(
+                "Rest {} for project [{}] will be copied to the test resources from the published jar (version: [{}]).",
+                getTestOrSpec(),
+                project.getPath(),
+                VersionProperties.getElasticsearch()
+            );
+            project.copy(c -> {
+                c.from(project.zipTree(coreConfig.getSingleFile()));
+                c.into(getTestSourceSet().getOutput().getResourcesDir()); // this ends up as the same dir as outputDir
+                c.include(includeCore.get().stream().map(prefix -> copyTo + "/" + prefix + "*/**").collect(Collectors.toList()));
+            });
+        }
+        // only include x-pack if explicitly instructed
+        if (includeXpack.get().isEmpty() == false) {
+            logger.info("X-pack YAML{} for project [{}] will be copied to the test resources.", getTestOrSpec(), project.getPath());
+            project.copy(c -> {
+                c.from(xpackConfig.getSingleFile());
+                c.into(getOutputDir());
+                c.include(xpackPatternSet.getIncludes());
+            });
+        }
+    }
+
+    /**
+     * Returns true if any files with a .yml extension exist the test resources rest-api-spec/test directory (from source or output dir)
+     */
+    private boolean projectHasYamlRestTests() {
```

Comment:
This should live in the plugin not in the task.

Reply:
Actually, thinking about this a bit more, I think this needs to stay in the task for a couple reasons:

1. This method checks against the source dir for tests committed to GH, but also checks the output directory for tests copied over by prior gradle tasks. This is why the spec task depends on the test task.  Since this evaluation is done in the task at execution time, I believe this is a reliable way to ensure that any source or copied tests exist. 

2. We can not actually skip the creation of the task based on the output of this. The task needs to execute if the project has no rest tests, but you requested to include some tests, and the IIUC the extension property isn't guaranteed to be resolved at configuration time. e.g. since we can not read the property at configuration time we will always need to create the task.